### ISSUE TITLE
Add capability to bake padding for the whole text into the vertex buffer

### ISSUE THREAD

**Issue by shdwcat** [JujuAdams/Scribble#190]
Currently, in my UI system, if I want to draw a scribble text element inside of the current `draw_rect` (aka the space that the text *could* fill), I do some math with the rect and the padding values to figure out the inner rect to use for scribble's `.wrap()` function. I then have to add the padding values back to the resulting scribble element's width/height in order to figure out how much of the draw_rect was actually used up by the text + padding.

For a few text elements this is no big deal, but for complex UIs with lots of buttons and tooltips and descriptions etc, all this little math adds up, so I was thinking it would be nice if scribble supported a `.padding(left, right, top, bottom)` function, that would bake the padding into the stored vertex buffer, and therefore relieving my code from having to do that math for text elements.

**Comment by JujuAdams** [comment]
Innnnnteresting. Which methods do you think padding would/should affect?

**Comment by shdwcat** [comment]
In my mind it would work like html padding:
![image](https://user-images.githubusercontent.com/15136382/144774078-16c583ff-6daf-4a90-a713-0aec29571808.png)
Padding goes *around* the entire content, but is part of the space the content takes up.

For example:
```
scribble()
    .wrap(300,300)
    .padding(10, 10, 10, 10) // or alternatively, allow padding(10) and interpret only one argument as 'use this for all 4)
```
Padding should go *inside* the wrap box, which means that the actual space for the text will be 280x280. And it means that if we call `draw(50,50)` then the actual text will be drawn at x:60 y:60. This also means that if you do `.get_bbox(50, 50)` those values will reflect the *padded* space, i.e. width will be 300 and height will be 300.

If we remove the `.wrap()` call, then do `draw(50,50)` we still have text drawn at 60,60. But the bbox width would now be the width of the drawn scribble text *plus* 10px padding on either side. So if the text would normally be 200 pixels wide, bbox width should now report 220.

Let say we add `align(fa_right)` and do `draw(800,800)`. Normally that means the right side of the text would end at x:800. With padding at 10, that means the right edge of the text should now hit 790. If the original text was 200px wide, then that means the left edge of the text should now bet at x:580 (800 - 220).

I think this should get the gist across? Let me know if you need more clarification!

**Comment by JujuAdams** [comment]
Implemented for v8.0.0 alpha 8
